include/itkPhaseCorrelationImageRegistrationMethod.h

```diff
@@ -94,13 +95,18 @@ namespace itk
  *  This class allows caching of image FFTs, because image montaging usually
  *  requires a single tile to participate in multiple image registrations.
  *
+ *  TInternalPixelTypePixel will be used by internal filters. It should be
+ *  float for integral and float inputs, and double for double inputs.
+ *
  * \author Jakub Bican, jakub.bican@matfyz.cz, Department of Image Processing,
  *         Institute of Information Theory and Automation,
  *         Academy of Sciences of the Czech Republic.
  *
  * \ingroup Montage
  */
-template< typename TFixedImage, typename TMovingImage >
+template< typename TFixedImage, typename TMovingImage,
+  typename TInternalPixelType = typename std::conditional<
```

Comment:
Same type logic should be applied in the example?

Reply:
In the library code, we don't know which type will be instantiated. In the examples we have concrete types.